File: LCObjective-C/LCObjective-C/Problems/PalindromeNumber/PalindromeNumber.m
//
//  PalindromeNumber.m
//  LCObjective-C
//
//  Created by Sherzod Akhmedov on 23/10/21.
//

#import <Foundation/Foundation.h>
#import "PalindromeNumber.h"

@implementation PalindromeNumber

+ (BOOL)isPalindrome:(int)n
{
    if (n < 0) {
        return false;
    }
    
    int m = n;
    int reversedNumber = 0;
    while (m != 0) {
        reversedNumber = reversedNumber * 10 + m % 10;
        m /= 10;
    }
    return reversedNumber == n;
}

@end

File: LCSwift/LCSwift/Problems/PalindromeNumber/PalindromeNumber.swift
//
//  PalindromeNumber.swift
//  LCSwift
//
//  Created by Sherzod Akhmedov on 23/10/21.
//

import Foundation

class PalindromeNumber {
    static func isPalindrome(_ x: Int) -> Bool {
        if x < 11 {
            return false
        }
        
        var y = x
        var reversedNumber: Int = 0
        
        while y != 0 {
            reversedNumber = reversedNumber * 10 + y % 10
            y /= 10
        }
        
        return reversedNumber == x
    }
}
